firmware/mcal/stm32f767/os/fifo.h

```diff
@@ -0,0 +1,51 @@
+/// @author Matteo Tullo
+/// @date 2024-02-24
+
+#pragma once
+
+#include <cstdint>
+
+#include "shared/util/os.h"
+#include "shared/os/fifo.h"
+#include "cmsis_os2.h"
+
+namespace mcal::os {
+
+
+// TODO: Add comments and handle errors more robustly
+class Fifo final : public shared::os::Fifo  {
+private:
+    osMessageQueueId_t* fifo_id;
```

Comment:
instance variables end in `_` https://google.github.io/styleguide/cppguide.html#Variable_Names